src/renderer/components/+preferences/add-helm-repo-dialog.tsx

```diff
@@ -0,0 +1,175 @@
+
+  @observable static isOpen = false;
+
+  static open() {
+    AddHelmRepoDialog.isOpen = true;
+  }
+
+  static close() {
+    AddHelmRepoDialog.isOpen = false;
+  }
+
+  @observable helmRepo : HelmRepo = this.emptyRepo;
+  @observable showOptions = false;
+
+  close = () => {
+    AddHelmRepoDialog.close();
+    this.helmRepo = this.emptyRepo;
+    this.showOptions = false;
+  };
+
+  setFilepath(type: FileType, value: string) {
+    this.helmRepo[type] = value;
+  }
+
+  getFilePath(type: FileType) : string {
+    return this.helmRepo[type];
+  }
+
+  async selectFileDialog(type: FileType, fileFilter: FileFilter) {
+    const { dialog, BrowserWindow } = remote;
+    const { canceled, filePaths } = await dialog.showOpenDialog(BrowserWindow.getFocusedWindow(), {
+      defaultPath: this.getFilePath(type),
+      properties: ["openFile", "showHiddenFiles"],
+      message: _i18n._(t`Select file`),
+      buttonLabel: _i18n._(t`Use file`),
+      filters: [
+        fileFilter,
+        { name: "Any", extensions: ["*"]}
+      ]
+    });
+
+    if (!canceled && filePaths.length) {
+      this.setFilepath(type, filePaths[0]);
+    }
+  }
+
+  async addCustomRepo() {
+    try {
+      await repoManager.addСustomRepo(this.helmRepo);
+      Notifications.ok(<Trans>Helm repository <b>{this.helmRepo.name}</b> has added</Trans>);
+      this.props.onAddRepo();
+      this.close();
+    } catch (err) {
+      Notifications.error(<Trans>Adding helm branch <b>{this.helmRepo.name}</b> has failed: {String(err)}</Trans>);
+    }
+  }
+
+  renderFileInput(placeholder:string, fileType:FileType ,fileExtensions:string[]){
+    return(
+      <div className="flex gaps align-center">
+        <Input
+          placeholder={placeholder}
+          validators = {isPath}
+          className="box grow"
+          value={this.getFilePath(fileType)}
+          onChange={v => this.setFilepath(fileType, v)}
+        />
+        <Icon
+          material="folder"
+          onClick={() => this.selectFileDialog(fileType, {name: placeholder, extensions: fileExtensions})}
+          tooltip={<Trans>Browse</Trans>}
+        />
+      </div>);
+  }
+
+  renderOptions() {
+    return (
+      <>
+        <SubTitle title={<Trans>Security settings</Trans>} />
+        <Checkbox
+          label={_i18n._(t`Skip TLS certificate checks for the repository`)}
+          value={this.helmRepo.insecureSkipTlsVerify}
+          onChange={v => this.helmRepo.insecureSkipTlsVerify = v}
+        />
+        {this.renderFileInput(_i18n._(t`Key file`), FileType.KeyFile, ["key", "keystore", "jks", "p12", "pfx", "pem"] )}
+        {this.renderFileInput(_i18n._(t`Ca file`), FileType.CaFile, ["crt", "cer", "ca-bundle", "p7b", "p7c" , "p7s", "p12", "pfx", "pem"] )}
+        {this.renderFileInput(_i18n._(t`Cerificate file`), FileType.CertFile, ["crt", "cer", "ca-bundle", "p7b", "p7c" , "p7s", "p12", "pfx", "pem"] )}
```

Comment:
I would like to see these extracted to `private static`'s as there is some duplication here.